### PR TITLE
Use value's semigroup instance more consistently across functions

### PR DESCRIPTION
**Pull request by 3noch**
Includes #36 and #37

### CODE REVIEW

**Comment by 3noch** [comment]
@bgamari Bump

File: Changelog.md [hunk 1]
```diff
@@ -1,9 +1,14 @@
 # Monoidal containers
 
+# 0.5.0.0
+
+  * Make `fromList`, `insert`, and `mapKeys` from `Data.Map.Monoidal` and `Data.Map.Monoidal.Strict` require `Semigroup` on values to properly capture monoidal behavior instead of reverting to the left-biased semantics of `Data.Map`.
+  * Support `these` 0.8.0
```
Comment (phadej) [inline Changelog.md]: fyi, there's `these-1` (and `semialign-1`) too now.

Comment (3noch) [reply Changelog.md]: Whew...well I guess we'll have to open another PR but at this rate we don't have to worry about those versions any time soon :joy:.

Comment (3noch) [reply Changelog.md]: @bgamari If you'd like help maintaining this project, I'd be happy to help. We use this library extensively.

**Comment by ali-abrar** [comment]
Running another travis build now that #37 has been merged.

**Pull request merged**